packages/vscode-extension/src/officeChat/telemetry.ts

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { LanguageModelChatMessage, LanguageModelChatMessageRole } from "vscode";
+import { ChatTelemetryData } from "../chat/telemetry";
+import { ITelemetryData } from "../chat/types";
+import { Correlator, getUuid } from "@microsoft/teamsfx-core";
+import { countMessagesTokens } from "../chat/utils";
+import {
+  TelemetryProperty,
+  TelemetrySuccess,
+  TelemetryTriggerFrom,
+} from "../telemetry/extTelemetryEvents";
+
+export enum OfficeChatTelemetryBlockReasonEnum {
+  RAI = "RAI",
+  OffTopic = "Off Topic",
+  LanguageModelError = "LanguageModel Error",
+}
+export class OfficeChatTelemetryData extends ChatTelemetryData {
```

Comment:
If you are implementing all member variables/methods, please implement the interface IChatTelemetryData instead of extend ChatTelemetryData

Reply:
will fix, thanks